scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,490 @@
+members:
+    - type: $t_structure_type_t
+      name: stype
+      desc: "[in] type of this structure"
+    - type: "void*"
+      name: pNext
+      desc: "[in,out][optional] must be null or a pointer to an extension-specific structure (i.e. contains stype and pNext)."
+      init: nullptr
+    - type: $t_value_t
+      name: "value"
+      desc: "[in] parameter value"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Query and get the available metric programmable handles."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetExp
+decl: static
+details:
+    - "Query the available programmable handles using *pCount = 0."
+    - "Returns all programmable metric handles available in the device."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: "$t_device_handle_t"
+      name: hDevice
+      desc: "[in] handle of the device"
+    - type: uint32_t*
+      name: pCount
+      desc: |
+            [in,out] pointer to the number of metric programmable handles.
+            if count is zero, then the driver shall update the value with the total number of metric programmable handles available.
+            if count is greater than the number of metric programmable handles available, then the driver shall update the value with the correct number of metric programmable handles available.
+    - type: "$t_metric_programmable_exp_handle_t*"
+      name: phMetricProgrammables
+      desc: |
+            [in,out][optional][range(0, *pCount)] array of handle of metric programmables.
+            if count is less than the number of metric programmables available, then driver shall only retrieve that number of metric programmables.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the properties of the metric programmable."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetPropertiesExp
+decl: static
+details:
+    - "Returns the properties of the metric programmable."
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: "$t_metric_programmable_exp_properties_t*"
+      name: pProperties
+      desc: "[out] properties of the metric programmable"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the information about the parameters of the metric programmable."
+version: "1.8"
+class: $tMetricProgrammable
+name: GetParamInfoExp
+decl: static
+details:
+    - "Returns information about the parameters of the metric programmable handle."
+params:
+    - type: $t_metric_programmable_exp_handle_t
+      name: hMetricProgrammable
+      desc: "[in] handle of the metric programmable"
+    - type: "uint32_t*"
+      name: pParameterCount
+      desc: |
+            [in,out] count of the parameters to retrieve parameter info.
```

Comment:
- missing the behavior of 0 to query the number. 
- What happens if the number is less than the actual available? 

General rules applicable for all the *count [in,out] params in the PR, since I found some comments apply to others:
- passing 0 to get the actual number
- passing a value smaller the available.
- passing a value bigger than available

Reply:
Here the usage of pParameterCount is slightly different:

Value of parameterCount is part of programmable properties ($t_metric_programmable_exp_properties_t).
So application has that information from `GetPropertiesExp`

Here the pParameterCount is for the application to specify the number of parameters it would like to query "information" for.

Similarly in the other APIs as well.

Reply:
OK, so only one case for in an out is needed. Clear